### PR TITLE
Implement test file serialization roundtrip

### PR DESCRIPTION
**Pull request by Ben0189**
Add unit tests for parseFile, stringifyFile, and transformDocument covering roundtrip fidelity and stability for all formats (yaml, toml, json, mdx), edge cases (empty files, unicode, malformed frontmatter), and $_body mapping - from transform document function.

### CODE REVIEW

File: packages/@tinacms/graphql/src/database/util.test.ts [hunk 1]
```diff
@@ -347,3 +348,188 @@ Content with components stored in frontmatter`;
...
+    const stringified1 = stringifyFile(withMeta(parsed), '.yaml', false);
+    const parsed2 = parseFile(stringified1, '.yaml', (yup) => yup.object({}));
+    const stringified2 = stringifyFile(withMeta(parsed2), '.yaml', false);
+
+    expect(stringified2).toEqual(stringified1);
+  });
+
+  it('parseFile throws on malformed YAML frontmatter in .md file', () => {
+    // unclosed bracket is invalid YAML frontmatter inside a .md file — gray-matter should throw, not silently return empty data
+    const malformed = `---\ntitle: [unclosed bracket\n---\n\nBody content.`;
```
Comment (18-th) [inline packages/@tinacms/graphql/src/database/util.test.ts:474]:
```suggestion
    const malformed = `---\ntitle: [unclosed bracket\n---\n\nBody content.`;
    const consoleError = vi
      .spyOn(console, 'error')
      .mockImplementation(() => {});
```

File: packages/@tinacms/graphql/src/database/util.test.ts [hunk 2]
```diff
@@ -347,3 +348,188 @@ Content with components stored in frontmatter`;
...
+
+    expect(() => {
+      parseFile(malformed, '.md', (yup) => yup.object({}));
+    }).toThrow();
```
Comment (18-th) [inline packages/@tinacms/graphql/src/database/util.test.ts:481]:
```suggestion
    }).toThrow();
    
    consoleError.mockRestore();
```

File: packages/@tinacms/graphql/src/database/util.test.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { describe, it, expect } from 'vitest';
```
Comment (18-th) [inline packages/@tinacms/graphql/src/database/util.test.ts]:
```suggestion
import { describe, it, expect, vi } from 'vitest';
```

**Review by 18-th — APPROVED** [review]
LGTM
